internal/pkg/cli/pipeline_init.go

```diff
@@ -220,38 +227,49 @@ func (o *initPipelineOpts) validateURL(url string) error {
 	return nil
 }
 
-func (o *initPipelineOpts) askEnvs() error {
-	if len(o.environments) == 0 {
-		envs, err := o.sel.Environments(pipelineSelectEnvPrompt, pipelineSelectEnvHelpPrompt, o.appName, func(order int) prompt.PromptConfig {
-			return prompt.WithFinalMessage(fmt.Sprintf("%s stage:", humanize.Ordinal(order)))
-		})
+// To avoid duplicating calls to GetEnvironment, validate and get config in the same step.
+func (o *initPipelineOpts) validateEnvsAndGetConfigs() error {
```

Comment:
it looks like this function and `getEnvConfigs()` are the same, could we remove one?